lightning/src/ln/peer_handler.rs

```diff
@@ -1179,6 +1179,18 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref> PeerManager<D
 		};
 	}
 
+	/// Disconnect a peer given its node id.
```

Comment:
Docs should indicate what `no_connection_possible` means (I believe there are other docs further up that describe it). Also should mention that reentrancy to disconnect_socket is likely.